### PR TITLE
Feature/save note

### PR DESCRIPTION
**Pull request by hmlampe**
note can be saved with extra button, but is also saved when toggling view

### CODE REVIEW

File: src/frontend/note.js [hunk 1]
```diff
@@ -27,6 +27,11 @@ class Note {
         }, false);
     }
 
+    #toggle_and_save(notelist) {
```
Comment (falk-werner) [inline src/frontend/note.js:30]: I prefer `toggle_preview` as name for this function. The fact, that it saves the current state of the editor is more of a side effect and not that important in my opinion.

```suggestion
    #toggle_preview(notelist) {
```

File: src/frontend/note.js [hunk 2]
```diff
@@ -104,6 +111,13 @@ class Note {
                 className: "fa fa-table",
                 title: "Insert Table"
             }, "|", {
+                name: "save",
+                action: () => { 
+                    notelist.save(this, this.#editor.value());
+                },
+                className: "fa fa-floppy-o no-disable",
```
Comment (falk-werner) [inline src/frontend/note.js:118]: I `no-disable`correct? The save button has no effect in preview mode.

```suggestion
                className: "fa fa-floppy-o",
```

File: src/frontend/notelist.js [hunk 3]
```diff
@@ -48,6 +48,10 @@ class NoteList {
         }
     }
 
+    async save(item, text) {
+        note.save(item.name, text);
```
Comment (falk-werner) [inline src/frontend/notelist.js:52]: Use the already existing `write` method instead creating a new one.

```suggestion
        note.write(item.name, text);
```

File: src/index.js [hunk 4]
```diff
@@ -25,6 +25,7 @@ app.on('ready', () => {
   ipcMain.handle('listNotes', note.list);
   ipcMain.handle('readNote', note.read);
   ipcMain.handle('createNote', note.create);
+  ipcMain.handle('saveNote', note.save);
```
Comment (falk-werner) [inline src/index.js:28]: Not needed - use `write`.

```suggestion
```

File: src/preload.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ contextBridge.exposeInMainWorld('note', {
     list: () => ipcRenderer.invoke('listNotes'),
     read: (name) => ipcRenderer.invoke('readNote', name),
     create: () => ipcRenderer.invoke('createNote'),
+    save: (name, text) => ipcRenderer.invoke('saveNote', name, text),
```
Comment (falk-werner) [inline src/preload.js:7]: Not needed - use `write`.

```suggestion
```

**Review by falk-werner — APPROVED** [review]
Nice feature. Please use already existing methods instead of re-inventing them.

**Review by falk-werner — CHANGES_REQUESTED** [review]
I was to fast to approve...

File: src/frontend/note.js (hunk 1 above)
Comment (hmlampe) [reply src/frontend/note.js]: commit war nicht ausreichend, da der Aufruf mit wechselt

**Pull request merged**